### ISSUE TITLE
60 fps videos

### ISSUE THREAD

**Issue by Ulrar** [trizen/youtube-viewer#67]
Hi,

Noticed the videos seems to always be at 30 fps, even if they play at 60 fps in the website player.
Is it even possible to get the videos at 60 with the method youtube-viewer uses ?

**Comment by trizen** [comment]
Hi Ulrar,

The answer is: "yes" and "no. youtube-viewer actually supports 60 fps videos, but they are supported only as a backup solution when a video at 30 fps is not available.

Here is the file where the itags are defined: https://github.com/trizen/youtube-viewer/blob/master/WWW-YoutubeViewer/lib/WWW/YoutubeViewer/Itags.pm#L76 (HFR = high frame rate), but, for each resolution, they are defined at the end of the list.

I will consider, in the future, adding an option such as `--prefer-hfr` to check for 60 fps videos first. Meanwhile, you can edit locally the Itags.pm file with the following content for '1080' and '720':

```perl
        '1080' => [
            [303,    # webm HFR (v)
             299,    # mp4 HFR (v)
            ],
            46,      # webm (v-a)
            37,      # mp4 (v-a)
            [248,    # webm (v)
             137,    # mp4 (v)
            ],
            96,      # ts (live) (v-a)
                  ],

          '720' => [
            [302,    # webm HFR (v)
             298,    # mp4 HFR (v)
            ],
            45,      # webm (v-a)
            22,      # mp4 (v-a)
            [247,    # webm (v)
             136,    # mp4 (v)
            ],

            120,     # flv (live) (v-a)
            95,      # ts (live) (v-a)
                   ],
```

**Comment by Ulrar** [comment]
Thank you for your answer, but I can't get that to work. I did edit the
file (/usr/lib64/perl5/vendor_perl/5.20.1/WWW/YoutubeViewer/Itags.pm on my
system) to replace the lines by that, but I still get 30 fps videos
according to mplayer :
ID_VIDEO_FPS=30.000

I use -identify in the mplayer args to check that. In my configuration I do
have resolution set to 720p.

Why choose 30fps as the default ? Bandwith concerns ? Could be an option in
the configuration file, just turn it on or of instead of an argument.

2015-04-04 15:25 GMT+02:00 Daniel Șuteu <notifications@github.com>:

> Hi Ulrar,
>
> The answer is: "yes" and "no. youtube-viewer actually supports 60 fps
> videos, but they are supported only as a backup solution when a video at 30
> fps is not available.
>
> Here is the file where the itags are defined:
> https://github.com/trizen/youtube-viewer/blob/master/WWW-YoutubeViewer/lib/WWW/YoutubeViewer/Itags.pm#L76
> (HFR = high frame rate), but, for each resolution, they are defined at the
> end of the list.
>
> I will consider, in the future, adding an option such as --prefer-hfr to
> check for 60 fps videos first. Meanwhile, you can edit locally the Itags.pm
> file with the following content for '1080' and '720':
>
>         '1080' => [
>             [303,    # webm HFR (v)
>              299,    # mp4 HFR (v)
>             ],
>             46,      # webm (v-a)
>             37,      # mp4 (v-a)
>             [248,    # webm (v)
>              137,    # mp4 (v)
>             ],
>             96,      # ts (live) (v-a)
>                   ],
>
>           '720' => [
>             [302,    # webm HFR (v)
>              298,    # mp4 HFR (v)
>             ],
>             45,      # webm (v-a)
>             22,      # mp4 (v-a)
>             [247,    # webm (v)
>              136,    # mp4 (v)
>             ],
>
>             120,     # flv (live) (v-a)
>             95,      # ts (live) (v-a)
>                    ],
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/trizen/youtube-viewer/issues/67#issuecomment-89577337>
> .
>



-- 
Kévin Lemonnier

**Comment by trizen** [comment]
The main concern was, indeed, the bandwidth, but nowadays the internet is pretty fast everywhere, and quality matters, so I think I will set 60 fps as default in the next release.

Try this video: https://www.youtube.com/watch?v=lfwjzNB--5k

**Comment by Ulrar** [comment]
Looking at the link it's getting the itag=22, so seems like I'm missing
something somewhere.

2015-04-07 21:17 GMT+02:00 Daniel Șuteu <notifications@github.com>:

> The main concern was, indeed, the bandwidth, but nowadays the internet is
> pretty fast everywhere, and quality matters, so I think I will set 60 fps
> as default in the next release.
>
> Try this video: https://www.youtube.com/watch?v=lfwjzNB--5k
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/trizen/youtube-viewer/issues/67#issuecomment-90704403>
> .
>



-- 
Kévin Lemonnier

**Comment by trizen** [comment]
Maybe DASH support is disabled? Try with command-line options `--dash --dash-m4a`. For more info, add the `--debug=2` option.

**Comment by Ulrar** [comment]
Okay that's my bad, I was using an outdated version. Looks like the package
in gentoo is still 3.1.1, I unmasked the git version and now it works fine.
I also had to compile in support for vp8 in ffmpeg but that's not really
related to youtube-viewer.

Thanks a lot, with the correct version and your modifications it works
perfectly ! :)

2015-04-07 23:55 GMT+02:00 Daniel Șuteu <notifications@github.com>:

> Maybe DASH support is disabled? Try with command-line options --dash
> --dash-m4a. For more info, add the --debug=2 option.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/trizen/youtube-viewer/issues/67#issuecomment-90743787>
> .
>



-- 
Kévin Lemonnier
